radicle-node/src/tests.rs

```diff
@@ -669,6 +669,96 @@ fn test_refs_announcement_relay() {
     );
 }
 
+#[test]
+fn test_refs_announcement_fetch_trusted_new_repo() {
+    logger::init(log::Level::Debug);
+
+    let tmp = tempfile::tempdir().unwrap();
+    let mut alice = Peer::config(
+        "alice",
+        [7, 7, 7, 7],
+        Storage::open(tmp.path().join("alice")).unwrap(),
+        peer::Config::default(),
+    );
+    let bob = {
+        let mut rng = fastrand::Rng::new();
+        let signer = MockSigner::new(&mut rng);
+        let storage = fixtures::storage(tmp.path().join("bob"), &signer).unwrap();
+
+        Peer::config(
+            "bob",
+            [9, 9, 9, 9],
+            storage,
+            peer::Config {
+                signer,
+                rng,
+                ..peer::Config::default()
+            },
+        )
+    };
+    let bob_inv = bob.storage().inventory().unwrap();
+    let rid = bob_inv[0];
+
+    alice.track_repo(&rid, tracking::Scope::Trusted).unwrap();
+    alice.connect_to(&bob);
+
+    // Alice receives Bob's refs.
+    alice.receive(bob.id(), bob.refs_announcement(rid));
+
+    // Alice fetches Bob's refs as this is a new repo.
+    assert_eq!(
+        alice.messages(bob.id()).next(),
+        Some(Message::Fetch { rid })
+    );
+}
+
+#[test]
+fn test_refs_announcement_fetch_trusted_negative() {
+    logger::init(log::Level::Debug);
+
+    let tmp = tempfile::tempdir().unwrap();
+    let bob = {
+        let mut rng = fastrand::Rng::new();
+        let signer = MockSigner::new(&mut rng);
+        let storage = fixtures::storage(tmp.path().join("bob"), &signer).unwrap();
+
+        Peer::config(
+            "bob",
+            [9, 9, 9, 9],
+            storage,
+            peer::Config {
+                signer,
+                rng,
+                ..peer::Config::default()
+            },
+        )
+    };
+    let bob_inv = bob.storage().inventory().unwrap();
+    let rid = bob_inv[0];
+
+    let mut alice = {
+        let mut rng = fastrand::Rng::new();
+        let signer = MockSigner::new(&mut rng);
+        let storage = fixtures::storage(tmp.path().join("alice"), &signer).unwrap();
+
+        // Create a repo on Alice using the same `rid`.
+        storage.create(rid).unwrap();
```

Comment:
This is my poor attempt to create the repo with the given `rid` on `alice`, before getting `refs_ann` from `bob`.  It is still not fulling working even the test passed, as there is `identity branch is missing` in the log. 

I believe there probably is a much simpler way but I didn't find it yet. My brain doesn't seem to work well with these tests. :(

Reply:
There is a `fixtures::project` that should create a new project properly. But as I mentioned elsewhere, it may not be needed.

Reply:
I've updated diff, and explained at [here ](https://github.com/radicle-dev/heartwood/pull/370#discussion_r1143973195).  For `fixtures::project` particularly, it creates a new project with a new Id, not a given Id. Hence I was not using it.